rdr_service/resource/generators/participant.py

```diff
@@ -594,59 +619,29 @@ def _make_stored_sample_dict_from_row(bss_row, has_order=True):
-        # See https://precisionmedicineinitiative.atlassian.net/browse/PDR-89
-        if stored_samples_added < ro_session.execute(_stored_samples_count_sql, {'bb_id': p_bb_id}).scalar():
-            # Include any "unknown order" samples associated with this participant
-            cursor = ro_session.execute(_samples_without_biobank_order_sql, {'bb_id': p_bb_id})
-            samples = [s for s in cursor]
-            if len(samples):
-                orders.append({'tests_ordered': 0, 'tests_stored': len(samples),
-                    'biobank_order_id': UNKNOWN_BIOBANK_ORDER_ID, 'samples': list()})
-                for row in samples:
-                    orders[-1]['samples'].append(_make_stored_sample_dict_from_row(row, has_order=False))
-
-        # Check to see that we have captured all of the stored samples for the orders.
-        # About 20% of the biobank orders have mis-matched ordered and stored sample counts.
-        for order in orders:
-            if order['tests_stored'] == 0 or order['tests_ordered'] == order['tests_stored']:
-                continue
-            # Fill in any missing ordered sample test records.
-            if order['tests_ordered'] > order['tests_stored'] and len(order['samples']) < order['tests_ordered']:
-                cursor = ro_session.execute(_biobank_ordered_samples_sql, {'order_id': order['biobank_order_id']})
-                results = [r for r in cursor]
-                existing_tests = [sample['test'] for sample in order['samples']]
-                for row in results:
-                    if row.test in existing_tests:
-                        continue
-                    order['samples'].append(_make_stored_sample_dict_from_row(row, has_order=True))
-            # Fill in any missing stored sample test records.
-            if order['tests_ordered'] < order['tests_stored']:
-                cursor = ro_session.execute(_biobank_stored_samples_sql, {'order_id': order['biobank_order_id']})
-                results = [r for r in cursor]
-                existing_tests = [sample['test'] for sample in order['samples']]
-                for row in results:
-                    if row.test in existing_tests:
-                        continue
-                    order['samples'].append(_make_stored_sample_dict_from_row(row, has_order=False))
+        # Add any "orderless" stored samples for this participant.  They will all be associated with a
+        # pseudo order with an order id of 'UNSET'
+        if len(bss_missing_orders):
+            orderless_stored_samples = list()
+            for bss_row in bss_missing_orders:
+                orderless_stored_samples.append(_make_sample_dict_from_row(bss=bss_row, bos=None))
+
+            orders.append({
+                'biobank_order_id': 'UNSET',
+                'tests_stored': len(orderless_stored_samples),
+                'samples': orderless_stored_samples
```

Comment:
I'm confused about some of the functionality here. Here (with the samples that don't have orders) the `samples` field is set to the list of stored samples. But up on line 624 the stored samples aren't put in the structure for orders that have samples, `samples` is set to an empty list. But in bq_participant_summary_dao.py, both structures (orders that have samples, and orderless samples) have the stored samples in the `samples` list. Is the difference intentional?

Reply:
That empty list was a bug.  Fixed in later commit